swift/Arrow/Sources/Arrow/ArrowDecoder.swift

```diff
@@ -126,11 +136,19 @@ private struct ArrowUnkeyedDecoding: UnkeyedDecodingContainer {
 
     mutating func decodeNil() throws -> Bool {
         defer {increment()}
-        return try self.decoder.doDecode(self.currentIndex) == nil
+        return try self.decoder.isNull(self.currentIndex)
     }
 
     mutating func decode<T>(_ type: T.Type) throws -> T where T: Decodable {
-        if type == Int8.self || type == Int16.self ||
+        if type == Int8?.self || type == Int16?.self ||
+            type == Int32?.self || type == Int64?.self ||
+            type == UInt8?.self || type == UInt16?.self ||
+            type == UInt32?.self || type == UInt64?.self ||
+            type == String?.self || type == Double?.self ||
+            type == Float?.self || type == Date?.self {
+            defer {increment()}
+            return try self.decoder.doDecode(self.currentIndex)!
+        } else if type == Int8.self || type == Int16.self ||
```

Comment:
I'm not familiar with Swift but can we unify this `if` and `else if`? It seems that body codes of them are same.

Reply:
They can be unified.  I had separated them to delineate the nullable and non nullable types but I can merge them.